src/service/url-replacements-impl.js

```diff
@@ -397,6 +397,37 @@ export class GlobalVariableSource extends VariableSource {
       return nav.userAgent;
     });
 
+    const HTML_ATTR_MAX_RETURN_SIZE = 10;
+    this.set('HTML_ATTR', (doc, cssSelector, ...attributeNames) => {
```

Comment:
I was doing a crapton of rebasing. This code has been moved into amp-analytics.js now.
But to answer your question: no, using this.ampdoc.win.document would get the publisher page, since all creatives share a global instance of url-replacements.
Moving it to amp-analytics gives us access to the doc object for the creative.